### PR TITLE
Add toml-sort hook

### PR DESCRIPTION
**Pull request by pmli**
This hook has more [configuration options](https://github.com/pappasam/toml-sort/tree/main?tab=readme-ov-file#configuration-file) than the one in #2125.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,196 +7,150 @@ name = "pymor"
 dynamic = ["version"]
 description = " "
 readme = "README.md"
-license.file = "LICENSE.txt"
+license = {file = "LICENSE.txt"}
```
Comment (sdrave) [inline pyproject.toml:10]: I don't get this change. 'Dotted keys' are part of the official toml spec and the dotted version is much easier to read. (I'm not even sure this is equivalent?)

**Review by sdrave — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (pmli) [reply pyproject.toml:10]: I based it on [this](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#license).

Comment (sdrave) [reply pyproject.toml:10]: Ok, but why change it? Apparently both versions are equivalent.

Comment (pmli) [reply pyproject.toml]: Ok, if they are equivalent, then I'm fine with either. It is not enforced by `toml-sort`. I reverted the change.

**Review by sdrave — DISMISSED** [review]

**Comment by sdrave** [comment]
@pmli, I don't get what toml-sort is doing here? Why does it move the 'license.file' entry to the end? (The entire section isn't really sorted, is it?)

**Comment by pmli** [comment]
> @pmli, I don't get what toml-sort is doing here? Why does it move the 'license.file' entry to the end? (The entire section isn't really sorted, is it?)

Yes, it seems like it moves and sorts entries with dots, and there is no option to disable it.

**Pull request merged**